### PR TITLE
Update to Scala 2.13

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
🙋‍♂️

[Scala 2.13 brings many good things!](https://github.com/scala/scala/releases/tag/v2.13.0)

I've had to update some dependencies to get it to work, but no large breaking changes.

### CODE REVIEW

**Comment by mthmulders** [comment]
Thanks a lot @hugo-vrijswijk for this PR! Seems promising to be able to use Scala 2.13 :-)

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     docker:
       - image: circleci/openjdk:8-jdk
         environment:
-          SBT_VERSION: 0.13.9
+          SBT_VERSION: 1.3.3
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply .circleci/config.yml]: No, but it was out of sync with the sbt version in build.properties. (also the new sbt version does have many performance improvements)

File: build.sbt [hunk 2]
```diff
@@ -1,18 +1,16 @@
 //
 // Define dependency versions
 //
-val akkaVer = "2.5.22"
-val akkaHttpVer = "10.1.8"
-val akkaSerialVer = "4.1.2"
+val akkaVer = "2.5.26"
+val akkaHttpVer = "10.1.10"
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply build.sbt]: Many of them are. 2.13 requires a new artifact publishing. So the latest version is usually needed. Not all of them were neccessary (like some Java libraries), but I figured while I was at it 🤷‍♂️

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,9 +1,9 @@
 resolvers += "Typesafe Repository" at "https://repo.typesafe.com/typesafe/releases/"
 
-addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.5.1")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.6.0")
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply project/plugins.sbt]: Yes, 2.13 support is included from 1.6.0

File: build.sbt [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
-val scalaMockVer = "3.6.0"
-val scalaTestVer = "3.0.7"
-val slickVer = "3.3.0"
-val sprayJsonVer = "1.3.3"
-val sprayWsVer = "0.1.4"
+val postgresqlVer = "42.2.8"
+val restAssuredVer = "4.1.2"
+val scalaMockVer = "4.4.0"
+val scalaTestVer = "3.0.8"
+val slickVer = "3.3.2"
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply build.sbt]: See above

File: project/plugins.sbt [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
-addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.0.9")
-addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.3.15")
-addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.4")
-addSbtPlugin("com.github.mwz" % "sbt-sonar" % "1.5.0")
+addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.0.10")
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.4.1")
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.5.0")
+addSbtPlugin("com.github.mwz" % "sbt-sonar" % "1.7.0")
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply project/plugins.sbt]: Some of them are (I think scapegoat and maybe sbt-sonar). If you want I can revert the change

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mthmulders) [reply .circleci/config.yml]: Excellent, we'll keep this one in, then :-)

File: project/plugins.sbt (hunk 3 above)
[the remark this answers is not in the document]
Comment (mthmulders) [reply project/plugins.sbt]: 👍

File: project/plugins.sbt (hunk 5 above)
[the remark this answers is not in the document]
Comment (mthmulders) [reply project/plugins.sbt]: We can keep them.

File: build.sbt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mthmulders) [reply build.sbt]: Hmm, I see. I usually prefer to have library upgrades one-by-one: in different PR's, like dependabot does for you. Unfortunately they don't support Scala / SBT yet :(.

[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply build.sbt]: You should have a look at [scala-steward](https://github.com/scala-steward/scala-steward)! We use it for Stryker4s and it works great. Even has [automated Scalafix migrations](https://github.com/fthomas/scala-steward/blob/master/docs/scalafix-migrations.md).

[the remark this answers is not in the document]
Comment (mthmulders) [reply build.sbt]: I'll do a bunch of lib upgrades first, just to make sure this PR gets as small as possible.

File: build.sbt (hunk 4 above)
[the remark this answers is not in the document]
Comment (mthmulders) [reply build.sbt]: See above :-)

File: build.sbt [hunk 6]
```diff
@@ -32,11 +30,9 @@ val parserComb        = "org.scala-lang.modules" %% "scala-parser-combinators"
 val postgresql        = "org.postgresql"         %  "postgresql"                  % postgresqlVer
 val restAssured       = "io.rest-assured"        %  "rest-assured"                % restAssuredVer
 val restAssuredScala  = "io.rest-assured"        %  "scala-support"               % restAssuredVer
-val scalaMock         = "org.scalamock"          %% "scalamock-scalatest-support" % scalaMockVer
+val scalaMock         = "org.scalamock"          %% "scalamock"                   % scalaMockVer
 val scalaTest         = "org.scalatest"          %% "scalatest"                   % scalaTestVer
 val slick             = "com.typesafe.slick"     %% "slick"                       % slickVer
-val sprayJson         = "io.spray"               %% "spray-json"                  % sprayJsonVer
```
Comment (hugo-vrijswijk) [inline build.sbt]: Am I correct in thinking this is not used anywhere and thus can be removed?

Comment (mthmulders) [reply build.sbt]: No, you're not correct. Spray is used in [HyperionJsonProtocol.scala](https://github.com/mthmulders/hyperion/blob/master/app/src/main/scala/hyperion/rest/HyperionJsonProtocol.scala) as well as [AppInfoService.scala](https://github.com/mthmulders/hyperion/blob/master/app/src/main/scala/hyperion/rest/AppInfoService.scala).

Could you please put it back?

Comment (hugo-vrijswijk) [reply build.sbt]: Since you asked nicely. The dependency wasn't actually added to the project, though. So I'll do that as well

**Pull request merged**